eclipse-theia/theia
# Enable fine-grained styling in code

#### What it does

Closes https://github.com/eclipse-theia/theia/issues/11063
Closes https://github.com/eclipse-theia/theia/issues/6743

Adds a new `StylingService` (and an associated `StylingParticipant`) which allows us to dynamically change styles using code. This has the advantage that we can set styles based on the existence of a color in a color theme.

As a simple example, we can take the notification hover color, which isn't set for some themes (hc-black). Hovering over a notification when the color isn't set will make the background transparent, while it should usually just keep the existing background color.

I used this feature to also implement the missing capabilities for high contrast theming and clean up some more (unused) css files (such as the `variables-dark.useable.css`/ `variables-bright.useable.css` files).

#### How to test

1. Open VSCode and Theia side by side
2. Select the dark high contrast theme for both of them
3. Assert that they behave (close to) the same with the high contrast theme
4. Assert that no visual regressions appear on other themes

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
